### PR TITLE
 Fix MessageInputBar bottomAnchor issue

### PR DESCRIPTION
**Pull request by zhongwuzw**

### CODE REVIEW

**Comment by zhongwuzw** [comment]
#357

File: Sources/Views/MessageInputBar.swift [hunk 1]
```diff
@@ -421,7 +421,7 @@ open class MessageInputBar: UIView {
     open func calculateIntrinsicContentSize() -> CGSize {
         
         let maxTextViewSize = CGSize(width: inputTextView.bounds.width, height: .greatestFiniteMagnitude)
-        var heightToFit = inputTextView.sizeThatFits(maxTextViewSize).height.rounded()
+        var heightToFit = inputTextView.sizeThatFits(maxTextViewSize).height.rounded() + padding.top + padding.bottom + textViewPadding.bottom + topStackView.bounds.height + bottomStackView.bounds.height
```
Comment (SD10) [inline Sources/Views/MessageInputBar.swift]: Can we move this into a local variable called `padding`?

File: Sources/Views/MessageInputBar.swift [hunk 2]
```diff
@@ -421,7 +421,7 @@ open class MessageInputBar: UIView {
     open func calculateIntrinsicContentSize() -> CGSize {
```
Comment (SD10) [inline Sources/Views/MessageInputBar.swift:426]: What about the `left` and `right` stack views? Their height can affect the total size of the input bar if their height is greater than the `inputTextView.height` & `topStackView.height`.

File: Sources/Views/MessageInputBar.swift [hunk 3]
```diff
@@ -421,7 +421,7 @@ open class MessageInputBar: UIView {
...
         
         let maxTextViewSize = CGSize(width: inputTextView.bounds.width, height: .greatestFiniteMagnitude)
-        var heightToFit = inputTextView.sizeThatFits(maxTextViewSize).height.rounded()
```
Comment (SD10) [inline Sources/Views/MessageInputBar.swift:424]: I think it should be `rounded(.up)` right?

File: Sources/Views/MessageInputBar.swift (hunk 2 above)
Comment (zhongwuzw) [reply Sources/Views/MessageInputBar.swift:426]: I think `left` or `right`'s `height` would not greater than that based on constraints.

File: Sources/Views/MessageInputBar.swift (hunk 3 above)
Comment (zhongwuzw) [reply Sources/Views/MessageInputBar.swift:424]: @SD10 ,yeah, default is `.toNearestOrAwayFromZero`, the `height` is not negative I think , so it's equal to `.up`?

**Comment by nathantannar4** [comment]
This used to be working. If you revert to my original PR that fixed this there is no issue. I will go back and revert some of the changes that were merged in that caused this.

**Comment by SD10** [comment]
Ok so what's going on here @zhongwuzw @nathantannar4?
This change was originally made 10 hours ago to development in #359 then reverted in #360. 
The original issue #357 was opened a few days ago before any of this

**Comment by zhongwuzw** [comment]
@SD10 @nathantannar4 ,the `master` branch also exists the same issue, it maybe look good in `default style` of inputBar(the reason it look good is the `intrinsicContentSize`,the [`master`](https://github.com/MessageKit/MessageKit/blob/389ccfa2b2b1c6c846e6ca27b2e1a57b4d8b02d8/Sources/Views/MessageInputBar.swift#L121) vs [`developer`](https://github.com/MessageKit/MessageKit/blob/f89307b78d7b44a43aa5dfd0f82d5b1dee9864b0/Sources/Views/MessageInputBar.swift#L424)), but we can see the issue apparently in `slack` `iMessage` style, like below:

**Comment by zhongwuzw** [comment]
@SD10 @nathantannar4 ,  [`PR`350](https://github.com/MessageKit/MessageKit/pull/350) causes the issue, It adjust the bottomStack.bottom's constraints, It break the layout system when keyboard appear , AutoLayout System would use `intrinsicContentSize` to get height .

**Comment by nathantannar4** [comment]
Damm ya I see that. I don't think `constraintLessThanOrEqualToSystemSpacingBelow(window.safeAreaLayoutGuide.bottomAnchor, multiplier: 1)` is working. That anchor seems to be off from what we need :( This is really frustrating

**Comment by nathantannar4** [comment]
I made some similar changes in #361.

**Comment by SD10** [comment]
We're going to close this in favor of #361

**Comment by zhongwuzw** [comment]
@SD10 , Yep, whatever, it's need to investigate in-depth, I just run the demo on iPhoneX, Based on [438f102](https://github.com/MessageKit/MessageKit/pull/361/commits/438f102ed280babed26e24b02fea7a9876c85ebd), it's not work correctly, besides that, some constraints ambiguous appear:



Capture some constraints ambiguous example:

```
Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<_UISystemBaselineConstraint:0x6000000bdbe0 MessageKit.InputStackView:0x7faa61d4b760.bottom <= UILayoutGuide:0x6000005be4c0'UIViewSafeAreaLayoutGuide'.bottom NSLayoutAnchorConstraintSpace(8)   (active)>",
    "<NSLayoutConstraint:0x60000009d510 UIInputSetContainerView:0x7faa61d507a0.bottom == UITextEffectsWindow:0x7faa62084400.bottom   (active)>",
    "<NSLayoutConstraint:0x60000009c610 V:[MessageKit.InputStackView:0x7faa61d49720]-(6)-[MessageKit.InputTextView:0x7faa620bf600]   (active)>",
    "<NSLayoutConstraint:0x60000009c2a0 MessageKit.InputTextView:0x7faa620bf600.bottom == MessageKit.InputStackView:0x7faa61d4b760.top   (active)>",
    "<NSLayoutConstraint:0x60000009bb20 V:|-(0)-[MessageKit.InputStackView:0x7faa61d49720]   (active, names: '|':MessageKit.MessageInputBar:0x7faa61c2b6e0 )>",
    "<NSLayoutConstraint:0x60000009cb60 MessageKit.InputTextView:0x7faa620bf600.height == 270.667   (active)>",
    "<NSLayoutConstraint:0x600000091a80 V:|-(0)-[MessageKit.MessageInputBar:0x7faa61c2b6e0]   (active, names: '|':UIInputSetHostView:0x7faa61c0e3d0 )>",
    "<NSLayoutConstraint:0x60400028e790 V:[UIInputSetContainerView:0x7faa61d507a0]-(0)-[UIInputSetHostView:0x7faa61c0e3d0]   (active)>",
    "<NSLayoutConstraint:0x60000009e4b0 'UIViewSafeAreaLayoutGuide-bottom' V:[UILayoutGuide:0x6000005be4c0'UIViewSafeAreaLayoutGuide']-(34)-|   (active, names: '|':UITextEffectsWindow:0x7faa62084400 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000009cb60 MessageKit.InputTextView:0x7faa620bf600.height == 270.667   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-11-29 16:28:54.391193+0800 ChatExample[65891:623712] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<_UISystemBaselineConstraint:0x6000000bdbe0 MessageKit.InputStackView:0x7faa61d4b760.bottom <= UILayoutGuide:0x6000005be4c0'UIViewSafeAreaLayoutGuide'.bottom NSLayoutAnchorConstraintSpace(8)   (active)>",
    "<NSLayoutConstraint:0x60000009d510 UIInputSetContainerView:0x7faa61d507a0.bottom == UITextEffectsWindow:0x7faa62084400.bottom   (active)>",
    "<NSLayoutConstraint:0x60000009c2a0 MessageKit.InputTextView:0x7faa620bf600.bottom == MessageKit.InputStackView:0x7faa61d4b760.top   (active)>",
    "<NSLayoutConstraint:0x60000009c8e0 V:[MessageKit.InputStackView:0x7faa61d49720]-(6)-[MessageKit.InputStackView:0x7faa61d4b560]   (active)>",
    "<NSLayoutConstraint:0x60000009ca20 MessageKit.InputStackView:0x7faa61d4b560.bottom == MessageKit.InputTextView:0x7faa620bf600.bottom   (active)>",
    "<NSLayoutConstraint:0x60000009bb20 V:|-(0)-[MessageKit.InputStackView:0x7faa61d49720]   (active, names: '|':MessageKit.MessageInputBar:0x7faa61c2b6e0 )>",
    "<NSLayoutConstraint:0x600000091a80 V:|-(0)-[MessageKit.MessageInputBar:0x7faa61c2b6e0]   (active, names: '|':UIInputSetHostView:0x7faa61c0e3d0 )>",
    "<NSLayoutConstraint:0x60400028e790 V:[UIInputSetContainerView:0x7faa61d507a0]-(0)-[UIInputSetHostView:0x7faa61c0e3d0]   (active)>",
    "<NSLayoutConstraint:0x60000009f680 'UISV-canvas-connection' MessageKit.InputStackView:0x7faa61d4b560.top <= MessageKit.InputBarButtonItem:0x7faa61d4b960'Send'.top   (active)>",
    "<NSLayoutConstraint:0x60000009f9a0 'UISV-canvas-connection' V:[MessageKit.InputBarButtonItem:0x7faa61d4b960'Send']-(0)-|   (active, names: '|':MessageKit.InputStackView:0x7faa61d4b560 )>",
    "<NSLayoutConstraint:0x60000009e4b0 'UIViewSafeAreaLayoutGuide-bottom' V:[UILayoutGuide:0x6000005be4c0'UIViewSafeAreaLayoutGuide']-(34)-|   (active, names: '|':UITextEffectsWindow:0x7faa62084400 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000009f680 'UISV-canvas-connection' MessageKit.InputStackView:0x7faa61d4b560.top <= MessageKit.InputBarButtonItem:0x7faa61d4b960'Send'.top   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-11-29 16:28:54.392307+0800 ChatExample[65891:623712] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<_UISystemBaselineConstraint:0x6000000bdbe0 MessageKit.InputStackView:0x7faa61d4b760.bottom <= UILayoutGuide:0x6000005be4c0'UIViewSafeAreaLayoutGuide'.bottom NSLayoutAnchorConstraintSpace(8)   (active)>",
    "<NSLayoutConstraint:0x60000009d510 UIInputSetContainerView:0x7faa61d507a0.bottom == UITextEffectsWindow:0x7faa62084400.bottom   (active)>",
    "<NSLayoutConstraint:0x60000009c610 V:[MessageKit.InputStackView:0x7faa61d49720]-(6)-[MessageKit.InputTextView:0x7faa620bf600]   (active)>",
    "<NSLayoutConstraint:0x60000009c2a0 MessageKit.InputTextView:0x7faa620bf600.bottom == MessageKit.InputStackView:0x7faa61d4b760.top   (active)>",
    "<NSLayoutConstraint:0x60000009bb20 V:|-(0)-[MessageKit.InputStackView:0x7faa61d49720]   (active, names: '|':MessageKit.MessageInputBar:0x7faa61c2b6e0 )>",
    "<NSLayoutConstraint:0x600000091a80 V:|-(0)-[MessageKit.MessageInputBar:0x7faa61c2b6e0]   (active, names: '|':UIInputSetHostView:0x7faa61c0e3d0 )>",
    "<NSLayoutConstraint:0x60400028e790 V:[UIInputSetContainerView:0x7faa61d507a0]-(0)-[UIInputSetHostView:0x7faa61c0e3d0]   (active)>",
    "<NSLayoutConstraint:0x60000009e4b0 'UIViewSafeAreaLayoutGuide-bottom' V:[UILayoutGuide:0x6000005be4c0'UIViewSafeAreaLayoutGuide']-(34)-|   (active, names: '|':UITextEffectsWindow:0x7faa62084400 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000009c610 V:[MessageKit.InputStackView:0x7faa61d49720]-(6)-[MessageKit.InputTextView:0x7faa620bf600]   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
..............

```

**Comment by SD10** [comment]
cc @nathantannar4

**Review by SD10 — APPROVED** [review]

**Pull request merged**